### PR TITLE
[master] Memory limit

### PR DESCRIPTION
**Pull request by mdouchement**
This PR aims to limit the memory usage of TSV Indexer.

- [x] Add ASCII schema of iterators
- [x] Try [CBOR](https://github.com/DamnWidget/cbor) encoding for performance improvements
- Improvements #27 (TODO after this PR)

### CODE REVIEW

**Comment by bmurzeau** [comment]
@mdouchement @wnkz following our investigations, the issue where the container gets killed when sorting a large dataset under memory constraint is due to:

_reason of the killed container_
```
[root@ip-10-103-3-41 ~]# docker inspect 7f5f4c740f3c | grep -i oom
            "OOMKilled": true,
            "OomScoreAdj": 0,
            "OomKillDisable": false,
```

_reason given by the Kernel_
```
[root@ip-10-103-3-41 ~]# tail /var/log/messages
Feb  8 14:15:47 ip-10-103-3-41 kernel: [77438.365485] [25028]     0 25028     1084       21       9       3        0             0 sh
Feb  8 14:15:47 ip-10-103-3-41 kernel: [77438.370329] [25029]     0 25029   827928   777553    1542       8        0             0 data-agent
Feb  8 14:15:47 ip-10-103-3-41 kernel: [77438.375281] Memory cgroup out of memory: Kill process 25029 (data-agent) score 990 or sacrifice child
Feb  8 14:15:47 ip-10-103-3-41 kernel: [77438.380228] Killed process 25029 (data-agent) total-vm:3311712kB, anon-rss:3110212kB, file-rss:0kB
```

This is due to an over consumption of RAM (ie if 2.5GB is allocated to the `data-agent`, it may consume more and therefore gets killed by the Kernel).

**Comment by mdouchement** [comment]
capture d ecran 2017-02-15 a 12 48 52

File: storage_service.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	}
+
+	return nil
+}
+
+// ------------------ //
+// MemStorageService  //
+// ------------------ //
+
+// A StorageService allows to reads and writes blobs on filesystem.
```
Comment (bmurzeau) [inline storage_service.go]: * typo

File: storage_service.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+type MemStorageService struct {
+	sync.RWMutex
+	compression diskv.Compression
+	registry    map[string][]byte
+	pool        *bpool.BufferPool
+}
+
+// NewStorageService inatanciates a new StorageService.
```
Comment (bmurzeau) [inline storage_service.go]: * typos

**Pull request merged**